include/ensmallen_bits/moead/weight_init_policies/uniform_init.hpp

```diff
@@ -0,0 +1,207 @@
+/**
+ * @file uniform_init.hpp
+ * @author Nanubala Gnana Sai
+ *
+ * The Uniform (Das Dennis) methodology of Weight Initialization.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_MOEAD_UNIFORM_HPP
+#define ENSMALLEN_MOEAD_UNIFORM_HPP
+
+namespace ens {
+
+/**
+ * The Uniform (Das Dennis) method for initializing weights. This algorithm guarantees
+ * that the distance between adjacent points would be uniform.
+ *
+ * For more information, see the following:
+ *
+ * @code
+ * article{zhang2007moea,
+ *   title={MOEA/D: A multiobjective evolutionary algorithm based on decomposition},
+ *   author={Zhang, Qingfu and Li, Hui},
+ *   journal={IEEE Transactions on evolutionary computation},
+ *   pages={712--731},
+ *   year={2007}
+ * @endcode
+ */
+class Uniform
+{
+ public:
+  /**
+   * Constructor for Uniform Weight Initializatoin Policy.
+   */
+  Uniform()
+  {
+    /* Nothing to do. */
+  }
+
+  /**
+   * Generate the reference direction matrix.
+   *
+   * @tparam MatType The type of the matrix used for constructing weights.
+   * @param numObjectives The dimensionality of objective space.
+   * @param numPoints The number of reference directions requested.
+   * @param epsilon Handle numerical stability after weight initialization.
+   */
+  template<typename MatType>
+  MatType Generate(size_t numObjectives,
+                   size_t numPoints,
+                   double epsilon)
+  {
+    size_t numPartitions  = FindNumParitions(numObjectives, numPoints);
+    size_t validNumPoints = FindNumUniformPoints(numObjectives, numPartitions);
+
+    //! The requested number of points is not matching any partition number.
+    if (numPoints != validNumPoints)
+    {
+      size_t nextValidNumPoints = FindNumUniformPoints(numObjectives, numPartitions + 1);
+      std::ostringstream oss;
```

Comment:
Slight hiccup, we actually need to use one of ensmallen's output streams:

Perhaps, we should use: 

```
std::ostream& Info
```

https://github.com/mlpack/ensmallen/blob/0af8004ff989ed4cce3a2bcaf7eeabb1313ac913/include/ensmallen_bits/log.hpp#L73-L77

Reply:
Thanks

Reply:
Okay so, I used  ```oss```  for throwing out error. Which should happen regardless of ```ENS_PRINT_INFO```.  So, once I put message inside ```oss```, I used ```std::logic_error(oss.str());```. Which I guess is the norm in our library.

Reply:
Agreed, I don't think this needs special handling, since this is for throwing an exception instead of outputting something, @coatless correct me if I'm wrong.

Reply:
@zoq you are correct. We do not have to worry about `oss` under `throw`. As an example, this pattern is used quite frequently elsewhere: 

https://github.com/mlpack/ensmallen/blob/1e70823989a67c89dc4909ff4152bd650b3abb77/include/ensmallen_bits/problems/logistic_regression_function_impl.hpp

Reply:
So, this doesn't need to change to using the `Info` portion. Whoops!